src/main/java/org/trigon/commands/GearRatioCalculationCommand.java

```diff
@@ -1,64 +1,130 @@
 package org.trigon.commands;
 
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import org.littletonrobotics.junction.Logger;
-import org.littletonrobotics.junction.networktables.LoggedDashboardBoolean;
 import org.littletonrobotics.junction.networktables.LoggedDashboardNumber;
+import org.trigon.hardware.phoenix6.cancoder.CANcoderEncoder;
+import org.trigon.hardware.phoenix6.cancoder.CANcoderSignal;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXMotor;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXSignal;
 
 import java.util.function.DoubleConsumer;
 import java.util.function.DoubleSupplier;
 
+/**
+ * A command that calculates and logs the gear ratio of a subsystem by comparing the distance traveled by a rotor and an encoder.
+ */
 public class GearRatioCalculationCommand extends Command {
     private final DoubleSupplier rotorPositionSupplier;
     private final DoubleSupplier encoderPositionSupplier;
     private final DoubleConsumer runGearRatioCalculation;
     private final String subsystemName;
-
+    private final double backlashAccountabilityTimeSeconds;
     private final LoggedDashboardNumber movementVoltage;
-    private final LoggedDashboardBoolean shouldMoveClockwise;
 
     private double startingRotorPosition;
     private double startingEncoderPosition;
     private double gearRatio;
+    private double startTime;
+    private boolean hasSetStartingPositions = false;
+
+    /**
+     * Creates a new GearRatioCalculationCommand.
+     * This constructor takes a motor to run the gear ratio calculation on, and an encoder to measure the distance traveled.
```

Comment:
mention this also takes the rotor position from the motor and checks its distance travelled